### DISCUSSION TITLE
Home assistant integration strikes again!

### DISCUSSION THREAD

**Discussion by atd** [OpenVoiceOS/ovos-buildroot#198] [Dev]
I continued the work of @mikejgray and resubmitted a PR to HomeAssistant

https://github.com/home-assistant/core/pull/103345

**Comment by mikejgray** [comment]
> I continued the work of @mikejgray and resubmitted a PR to HomeAssistant
> 
> 
> 
> https://github.com/home-assistant/core/pull/103345

Much appreciated - I never got the hang of the config stuff! I know @JarbasAl is doing some cool stuff with his Home Assistant that will eventually make it out into the world. He's using his HiveMind project to connect HA to the OVOS bus instead of going directly.

**Comment by atd** [comment]
I would be great to see that! I am sure I would learn a lot

My first goal is being able to get and set volume levels and night mode, so I can schedule them from HA

**Comment by JarbasAl** [comment]
on the HA front I have some WIP projects to share 
- https://github.com/JarbasHiveMind/hivemind-home-assistant-notify - secure connections to the bus, will likely replace the ovos integration at some point, PRs welcome!
- https://github.com/OpenVoiceOS/ovos-PHAL-sensors - OVOS device sensors in HA
- https://github.com/OpenVoiceOS/ovos-PHAL-zmqamera - OVOS camera in HA  (with object detection sensors)

**Comment by atd** [comment]
The home assistant PR is stale, but I continued with some work:

* Volume number
* Brightness number
* Gui Page select (created needed events https://github.com/OpenVoiceOS/skill-ovos-homescreen/pull/98 )
* Utterance text

You can check them at https://github.com/atd/core

The goal I wanted to achieve is setting volume 0, brightness 0.1 and night time page when we go to sleep, and volume 0.5, brightness 0.5 and main page when we wake up

Also, having utterances in HA so I can play with STT engines

**Reply by JarbasAl** [reply]
have a look at the OVOS sensors package, i provide a lot of system stuff in there via MQTT to home assistant

volume and brightness should be there, i use the pulseaudio sensors to know what music is playing and where, for example i can detect
- bluetooth connected -> stereo is in use
- pulse audio now playing string -> detect netflix/spotify in use

I might have missed the volume control :thinking: 


In general I want to add A LOT of bus based sensors, have the base class here https://github.com/OpenVoiceOS/ovos-PHAL-sensors/blob/dev/ovos_PHAL_sensors/sensors/base.py#L98 just need to start implementing more sensors

**Reply by atd** [reply]
what are the pros / cons of using MQTT vs the direct websocket connection?

**Reply by JarbasAl** [reply]
direct websocket connection is **STRONGLY** recommended against, it is the major security hole for OVOS and the port should be firewalled all the time. HiveMind is the near future solution for this approach as it gets you isolation, authentication and encryption to the bus, at the expense of needing more configuration per connection. what you did in your integration could use hivemind and work in the same way if relevant bus messages are authorized in hivemind. I might steal a bunch of your code for the hivemind-notify repo ;) (or PRs welcomed!)

for me the advantage of the other approach is that it is more portable across IOT platforms, in ovos-sensors the logging of sensor state changes is abstracted away and multiple frameworks can be enabled, HA HTTP and MQTT are just 2 possible SensorLoggers

most notably, ovos-sensors emits sensor readings to the messagebus which will integrate with OVOS CommonIOT during 0.0.9 dev cycle
